onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -257,6 +257,11 @@ class PlannerImpl {
 
   // Find if there exists some input tensor that we can use in-place for output_arg_num-th input in the node.
   bool FindReusableInput(const onnxruntime::Node& node, int output_arg_num, OrtValueIndex* reusable_input) {
+    auto p_next_node = node.OutputNodesBegin();
+    if (p_next_node != node.OutputNodesEnd() && p_next_node->OpType() == "YieldOp") {
+      return false;
+    }
```

Comment:
This does not handle the scenario where tensor can be consumed by multiple nodes not just Yield. You will need to add a copy node for Yield in this scenario.

Reply:
As discussed offline, we will do the conditional copy for fetches in PartialGraphExecutor in a separated PR.